keps/sig-node/2837-pod-level-resource-spec/README.md

```diff
@@ -0,0 +1,1458 @@
+
+      * `pod.spec.resources.limits[memory/cpu] < sum(pod.spec.containers[*].resources.limits[memory/cpu]) + sum(pod.spec.initContainers[*].resources.limits[memory/cpu])` is valid.
+        
+        For example: consider a pod with multiple containers:
+          1. Container 1: Moderate CPU and memory usage, but occasional spikes.
+          2. Container 2: Low and consistent resource usage.
+
+
+          ```yaml
+          apiVersion: v1
+          kind: Pod
+          ...
+          spec:
+            resources:
+              requests:
+                memory: 100Mi
+                cpu: 1
+              limits:
+                memory: 400Mi
+                cpu: 2.5
+            containers:
+            - name: container-1
+                resources:
+                  requests:
+                    memory: 60Mi
+                    cpu: 0.5
+                  limits:
+                    memory: 300Mi
+                    cpu: 2  
+            - name: container-2
+                resources:
+                  requests:
+                    memory: 40Mi
+                    cpu: 0.5
+                  limits:
+                    memory: 200Mi
+                    cpu: 1 
+          ```
+
+          The pod-level limit will serve as an aggregate ceiling. This will cover the scenario where the sum of peak usage for individual containers in a pod is high, but these
+          peaks don’t necessarily occur simultaneously.
+
+          If we simply set the pod-level limit as the sum of the individual container limits, we might overallocate resources. The pod might never actually use all those resources
+          at once, leading to wasted capacity on the node.
+
+      * `pod.spec.resources.limits[memory/cpu] < pod.spec.containers[i].resources.limits[memory/cpu]` is invalid.
+
+        A single container’s limit should not be allowed to exceed the pod level limit.
+  4. Validation logic comparing requests and limits
+
+      1. Pod level requests greater than pod level limits is invalid.
+
+      2. Pod level requests greater than sum of limits of all containers is invalid.
+
+      3. Sum of container level requests greater than Pod level limits is invalid.
+        
+          This is already ensured by other validation rules in 1. and 2. as following:
+        
+          As per 1. Container level Requests <= Pod level requests is valid<br>
+          And as per 3.1 Pod-level requests <= Pod level limits is valid<br>
+          Hence, by transitivity these rules will ensure Container level requests <= Pod-level limits.
+
+      4. [Already exists] Sum of container level requests less than container level limits is invalid.
+
+#### Init Containers & Sidecar Containers
+
+With the current implementation (before this KEP), Pod’s effective resource request/limit is the sum of 
+pod overhead and the higher of:
+
+* The sum of all non-init containers(app and sidecar containers) request/limit for a resource.
```

Comment:
the formulae is more complicated than this as sidecars are running while init containers are executed. You can link here: https://github.com/kubernetes/enhancements/blob/master/keps/sig-node/753-sidecar-containers/README.md#resources-calculation-for-scheduling-and-pod-admission